### PR TITLE
Fixed crashes on some SOAP 1.2 requests when using GetBufferedInputStream

### PR DESCRIPTION
**Pull request by theiji**
This fixes issue [1759 ](https://github.com/elastic/apm-agent-dotnet/issues/1759) crashes on some SOAP 1.2 requests when using GetBufferedInputStream

### CODE REVIEW

**Comment by z1c0** [comment]
Thanks for your contribution @theiji!
May I ask you to add a reproducer/test to your PR, please?
You can find existing tests for SOAP requests [here](https://github.com/elastic/apm-agent-dotnet/blob/main/test/Elastic.Apm.AspNetFullFramework.Tests/Soap/SoapRequestTests.cs).

**Comment by z1c0** [comment]
@theiji I was able to verify your fix - thanks again!
I think I could simply the code a little bit and added an additional test (small & large payload).
May I ask to to update your PR the way I suggested in my branch https://github.com/elastic/apm-agent-dotnet/tree/1759-bug-elasticapmaspnetfullframework-crashes-on-some-soap-requests-to-asp-net-web-service?

**Comment by z1c0** [comment]
FYI @gregkalapos - I was able to reproduce the issue and verify the fix provided here.
The underlying problem is that in [GetSoap12ActionFromInputStream](https://github.com/elastic/apm-agent-dotnet/blob/20fea7888fd3a044e13fb34c3e657d26502a6911/src/Elastic.Apm.AspNetFullFramework/Extensions/SoapRequest.cs#L85) we use an `XmlReader` to read the posted payload data. For large payloads, we end up in a situation where the `XmlReader` only reads part of the stream (as large as its internal buffer size). 

image

This leaves the underlying stream in a "half-read" state that, later on,  causes the internal server error. By making sure that the stream is read to the end at all times, this issue is fixed.

**Review by z1c0 — APPROVED** [review]
LGTM!
(It looks like you still need to sign the CLA though @theiji)

**Review by gregkalapos — APPROVED** [review]
LGTM as well.

@theiji please make sure you use the same e-mail address that you used for your commits when you sign the CLA - [it’s here in the 2. line](https://github.com/elastic/apm-agent-dotnet/commit/fe639a0cee39dab39067f0be855e82d6687ff033.patch).

We can merge once the CLA is signed.

**Comment by theiji** [comment]
@gregkalapos sorry. Can I change e-mail address in this commits. My profile on notebook was wrong to commit. The correct is theiji@hotmail.com

File: test/Elastic.Apm.AspNetFullFramework.Tests/Soap/SoapRequestTests.cs [hunk 1]
```diff
@@ -26,12 +26,15 @@ public SoapRequestTests(ITestOutputHelper xUnitOutputHelper)
 		/// to parse the parameters for the web method.
 		/// </summary>
 		[AspNetFullFrameworkFact]
```
Comment (gregkalapos) [inline test/Elastic.Apm.AspNetFullFramework.Tests/Soap/SoapRequestTests.cs:28]:
```suggestion
		[AspNetFullFrameworkTheory]
```

Currently it fails:

```
System.InvalidOperationException : [Fact] methods are not allowed to have parameters. Did you mean to use [Theory]?
```

Due to the new parameter, it needs to be `AspNetFullFrameworkTheory`

**Review by gregkalapos — CHANGES_REQUESTED** [review]
Thanks for signing the CLA @theiji.

There is one minor issue with the changed test - details below. Sorry for missing that previously; I just noticed that after CI ran the PR (it wasn't running the PR previously due to the missing CLA).

**Comment by theiji** [comment]
@gregkalapos done

**Comment by z1c0** [comment]
Thanks for hanging on @theiji - merge is imminent :-)

**Review by gregkalapos — APPROVED** [review]
Thanks for your contribution @theiji!

**Pull request merged**